### PR TITLE
Python.d/postgres.chart.py PEP 8 code cleanup

### PR DESCRIPTION
**Pull request by Ferroin**
I've split this out as a separate PR from the other module PR's because it will require some work that I'm not confident I can do myself (and because the changes I can do cover a _huge_ percentage of the file).  What's currently in this branch is coding-style compliant _except_ for the SQL code embedded in the module.

There are a total of 6 locations within the SQL code where a line is longer than 120 characters (located on lines 115, 210-213, and 258).  While I have basic skills with SQL, this code is complex enough that I'm pretty certain I'll break something if I change it (and I can't really test it easily since I don't have any PostgreSQL deployments I can run it against).

Relevant to: #4167

### CODE REVIEW

**Comment by paulfantom** [comment]
There is a tool sqllint.com which can beautify SQL code. Here is a patch:

```diff
diff --git a/python.d/postgres.chart.py b/python.d/postgres.chart.py
index 0b042fde..34ee8cac 100644
--- a/python.d/postgres.chart.py
+++ b/python.d/postgres.chart.py
@@ -106,135 +106,225 @@ METRICS = {
 QUERIES = {
     'WAL': """
 SELECT
-  count(*) as total_wal,
-  count(*) FILTER (WHERE type = 'recycled') AS recycled_wal,
-  count(*) FILTER (WHERE type = 'written') AS written_wal
+    count(*) as total_wal,
+    count(*) FILTER (
+WHERE
+    type = 'recycled') AS recycled_wal,   count(*) FILTER (
+WHERE
+    type = 'written') AS written_wal 
 FROM
-  (SELECT wal.name,
-     pg_{0}file_name(CASE pg_is_in_recovery() WHEN true THEN NULL ELSE pg_current_{0}_{1}() END ),
-     CASE WHEN wal.name > pg_{0}file_name(CASE pg_is_in_recovery() WHEN true THEN NULL ELSE pg_current_{0}_{1}() END ) THEN 'recycled'
-       ELSE 'written'
-     END AS type
-  FROM pg_catalog.pg_ls_dir('pg_{0}') AS wal(name)
-  WHERE name ~ '^[0-9A-F]{{24}}$'
-  ORDER BY (pg_stat_file('pg_{0}/'||name)).modification, wal.name DESC) sub;
+    (SELECT
+        wal.name,
+        pg_{0}file_name(CASE pg_is_in_recovery() 
+            WHEN true THEN NULL 
+            ELSE pg_current_{0}_{1}() 
+        END ),
+        CASE 
+            WHEN wal.name > pg_{0}file_name(CASE pg_is_in_recovery() 
+                WHEN true THEN NULL 
+                ELSE pg_current_{0}_{1}() 
+            END ) THEN 'recycled'        
+            ELSE 'written'      
+        END AS type   
+    FROM
+        pg_catalog.pg_ls_dir('pg_{0}') AS wal(name)   
+    WHERE
+        name ~ '^[0-9A-F]{{24}}$'   
+    ORDER BY
+        (pg_stat_file('pg_{0}/'||name)).modification,
+        wal.name DESC) sub;
 """,
     'ARCHIVE': """
 SELECT
     CAST(COUNT(*) AS INT) AS file_count,
-    CAST(COALESCE(SUM(CAST(archive_file ~ $r$\.ready$$r$ as INT)), 0) AS INT) AS ready_count,
-    CAST(COALESCE(SUM(CAST(archive_file ~ $r$\.done$$r$ AS INT)), 0) AS INT) AS done_count
+    CAST(COALESCE(SUM(CAST(archive_file ~ $r$\.ready$$r$ as INT)),
+    0) AS INT) AS ready_count,
+    CAST(COALESCE(SUM(CAST(archive_file ~ $r$\.done$$r$ AS INT)),
+    0) AS INT) AS done_count 
 FROM
     pg_catalog.pg_ls_dir('pg_{0}/archive_status') AS archive_files (archive_file);
 """,
     'BACKENDS': """
 SELECT
-    count(*) - (SELECT count(*) FROM pg_stat_activity WHERE state = 'idle') AS backends_active,
-    (SELECT count(*) FROM pg_stat_activity WHERE state = 'idle' ) AS backends_idle
-FROM  pg_stat_activity;
+    count(*) - (SELECT
+        count(*) 
+    FROM
+        pg_stat_activity 
+    WHERE
+        state = 'idle') AS backends_active,
+    (SELECT
+        count(*) 
+    FROM
+        pg_stat_activity 
+    WHERE
+        state = 'idle' ) AS backends_idle 
+FROM
+    pg_stat_activity;
 """,
     'TABLE_STATS': """
 SELECT
-  ((sum(relpages) * 8) * 1024) AS table_size,
-  count(1)                     AS table_count
-FROM pg_class
-WHERE relkind IN ('r', 't');
+    ((sum(relpages) * 8) * 1024) AS table_size,
+    count(1)                     AS table_count 
+FROM
+    pg_class 
+WHERE
+    relkind IN (
+        'r', 't'
+    );
 """,
     'INDEX_STATS': """
 SELECT
-  ((sum(relpages) * 8) * 1024) AS index_size,
-  count(1)                     AS index_count
-FROM pg_class
-WHERE relkind = 'i';""",
-    'DATABASE': """
-SELECT
-  datname AS database_name,
-  numbackends AS connections,
-  xact_commit AS xact_commit,
-  xact_rollback AS xact_rollback,
-  blks_read AS blks_read,
-  blks_hit AS blks_hit,
-  tup_returned AS tup_returned,
-  tup_fetched AS tup_fetched,
-  tup_inserted AS tup_inserted,
-  tup_updated AS tup_updated,
-  tup_deleted AS tup_deleted,
-  conflicts AS conflicts,
-  pg_database_size(datname) AS size,
-  temp_files AS temp_files,
-  temp_bytes AS temp_bytes
-FROM pg_stat_database
-WHERE datname IN %(databases)s
-;
+    ((sum(relpages) * 8) * 1024) AS index_size,
+    count(1)                     AS index_count 
+FROM
+    pg_class 
+WHERE
+    relkind = 'i';
+""",
+'DATABASE': """ SELECT
+    datname AS database_name,
+    numbackends AS connections,
+    xact_commit AS xact_commit,
+    xact_rollback AS xact_rollback,
+    blks_read AS blks_read,
+    blks_hit AS blks_hit,
+    tup_returned AS tup_returned,
+    tup_fetched AS tup_fetched,
+    tup_inserted AS tup_inserted,
+    tup_updated AS tup_updated,
+    tup_deleted AS tup_deleted,
+    conflicts AS conflicts,
+    pg_database_size(datname) AS size,
+    temp_files AS temp_files,
+    temp_bytes AS temp_bytes 
+FROM
+    pg_stat_database 
+WHERE
+    datname IN %(
+        databases
+    )s ;
 """,
     'BGWRITER': """
 SELECT
-  checkpoints_timed AS checkpoint_scheduled,
-  checkpoints_req AS checkpoint_requested,
-  buffers_checkpoint * current_setting('block_size')::numeric buffers_checkpoint,
-  buffers_clean * current_setting('block_size')::numeric buffers_clean,
-  maxwritten_clean,
-  buffers_backend * current_setting('block_size')::numeric buffers_backend,
-  buffers_alloc * current_setting('block_size')::numeric buffers_alloc,
-  buffers_backend_fsync
-FROM pg_stat_bgwriter;
+    checkpoints_timed AS checkpoint_scheduled,
+    checkpoints_req AS checkpoint_requested,
+    buffers_checkpoint * current_setting('block_size')::numeric buffers_checkpoint,
+    buffers_clean * current_setting('block_size')::numeric buffers_clean,
+    maxwritten_clean,
+    buffers_backend * current_setting('block_size')::numeric buffers_backend,
+    buffers_alloc * current_setting('block_size')::numeric buffers_alloc,
+    buffers_backend_fsync 
+FROM
+    pg_stat_bgwriter;
 """,
     'LOCKS': """
 SELECT
-  pg_database.datname as database_name,
-  mode,
-  count(mode) AS locks_count
-FROM pg_locks
-  INNER JOIN pg_database ON pg_database.oid = pg_locks.database
-GROUP BY datname, mode
-ORDER BY datname, mode;
+    pg_database.datname as database_name,
+    mode,
+    count(mode) AS locks_count 
+FROM
+    pg_locks   
+INNER JOIN
+    pg_database 
+        ON pg_database.oid = pg_locks.database 
+GROUP BY
+    datname,
+    mode 
+ORDER BY
+    datname,
+    mode;
 """,
     'FIND_DATABASES': """
-SELECT datname
-FROM pg_stat_database
-WHERE has_database_privilege((SELECT current_user), datname, 'connect')
-AND NOT datname ~* '^template\d+';
+SELECT
+    datname 
+FROM
+    pg_stat_database 
+WHERE
+    has_database_privilege((SELECT
+        current_user), datname, 'connect') 
+    AND NOT datname ~* '^template\d ';
 """,
     'FIND_STANDBY': """
-SELECT application_name
-FROM pg_stat_replication
-WHERE application_name IS NOT NULL
-GROUP BY application_name;
+SELECT
+    application_name 
+FROM
+    pg_stat_replication 
+WHERE
+    application_name IS NOT NULL 
+GROUP BY
+    application_name;
 """,
     'FIND_REPLICATION_SLOT': """
 SELECT slot_name
 FROM pg_replication_slots;
 """,
     'STANDBY_DELTA': """
-SELECT application_name,
-  pg_{0}_{1}_diff(CASE pg_is_in_recovery() WHEN true THEN pg_last_{0}_receive_{1}() ELSE pg_current_{0}_{1}() END , sent_{1}) AS sent_delta,
-  pg_{0}_{1}_diff(CASE pg_is_in_recovery() WHEN true THEN pg_last_{0}_receive_{1}() ELSE pg_current_{0}_{1}() END , write_{1}) AS write_delta,
-  pg_{0}_{1}_diff(CASE pg_is_in_recovery() WHEN true THEN pg_last_{0}_receive_{1}() ELSE pg_current_{0}_{1}() END , flush_{1}) AS flush_delta,
-  pg_{0}_{1}_diff(CASE pg_is_in_recovery() WHEN true THEN pg_last_{0}_receive_{1}() ELSE pg_current_{0}_{1}() END , replay_{1}) AS replay_delta
-FROM pg_stat_replication
-WHERE application_name IS NOT NULL;
+SELECT
+    application_name,
+    pg_{0}_{1}_diff(CASE pg_is_in_recovery() 
+        WHEN true THEN pg_last_{0}_receive_{1}() 
+        ELSE pg_current_{0}_{1}() 
+    END ,
+    sent_{1}) AS sent_delta,
+    pg_{0}_{1}_diff(CASE pg_is_in_recovery() 
+        WHEN true THEN pg_last_{0}_receive_{1}() 
+        ELSE pg_current_{0}_{1}() 
+    END ,
+    write_{1}) AS write_delta,
+    pg_{0}_{1}_diff(CASE pg_is_in_recovery() 
+        WHEN true THEN pg_last_{0}_receive_{1}() 
+        ELSE pg_current_{0}_{1}() 
+    END ,
+    flush_{1}) AS flush_delta,
+    pg_{0}_{1}_diff(CASE pg_is_in_recovery() 
+        WHEN true THEN pg_last_{0}_receive_{1}() 
+        ELSE pg_current_{0}_{1}() 
+    END ,
+    replay_{1}) AS replay_delta 
+FROM
+    pg_stat_replication 
+WHERE
+    application_name IS NOT NULL;
 """,
     'REPSLOT_FILES': """
-WITH wal_size AS (
-    SELECT current_setting('wal_block_size')::INT * setting::INT AS val
-    FROM pg_settings
-    WHERE name = 'wal_segment_size'
-)
-SELECT slot_name, slot_type, replslot_wal_keep, count(slot_file) AS replslot_files
-FROM (
-  SELECT slot.slot_name, CASE WHEN slot_file <> 'state' THEN 1 END AS slot_file , slot_type,
-    COALESCE (floor((pg_wal_lsn_diff (pg_current_wal_lsn (),
-              slot.restart_lsn) - (pg_walfile_name_offset (restart_lsn)).file_offset) / (s.val)),
-            0) AS replslot_wal_keep
-  FROM pg_replication_slots slot
-  LEFT JOIN (
-    SELECT slot2.slot_name,
-           pg_ls_dir('pg_replslot/' || slot2.slot_name) AS slot_file
-      FROM pg_replication_slots slot2
-    ) files (slot_name, slot_file)
-   ON slot.slot_name = files.slot_name
-  CROSS JOIN wal_size s) AS d
-GROUP BY slot_name, slot_type, replslot_wal_keep;
+WITH wal_size AS (     SELECT
+    current_setting('wal_block_size')::INT * setting::INT AS val     
+FROM
+    pg_settings     
+WHERE
+    name = 'wal_segment_size' ) SELECT
+    slot_name,
+    slot_type,
+    replslot_wal_keep,
+    count(slot_file) AS replslot_files 
+FROM
+    (   SELECT
+        slot.slot_name,
+        CASE 
+            WHEN slot_file <> 'state' THEN 1 
+        END AS slot_file ,
+        slot_type,
+        COALESCE (floor((pg_wal_lsn_diff (pg_current_wal_lsn (),
+        slot.restart_lsn) - (pg_walfile_name_offset (restart_lsn)).file_offset) / (s.val)),
+        0) AS replslot_wal_keep   
+    FROM
+        pg_replication_slots slot   
+    LEFT JOIN
+        (
+            SELECT
+                slot2.slot_name,
+                pg_ls_dir('pg_replslot/' || slot2.slot_name) AS slot_file       
+            FROM
+                pg_replication_slots slot2     
+        ) files (slot_name, slot_file)    
+            ON slot.slot_name = files.slot_name   CROSS 
+    JOIN
+        wal_size s
+    ) AS d 
+GROUP BY
+    slot_name,
+    slot_type,
+    replslot_wal_keep;
 """,
     'IF_SUPERUSER': """
 SELECT current_setting('is_superuser') = 'on' AS is_superuser;
@@ -244,18 +334,31 @@ SHOW server_version_num;
 """,
     'AUTOVACUUM': """
 SELECT
-  count(*) FILTER (WHERE query LIKE  'autovacuum: ANALYZE%%') AS analyze,
-  count(*) FILTER (WHERE query LIKE  'autovacuum: VACUUM ANALYZE%%') AS vacuum_analyze,
-  count(*) FILTER (WHERE query LIKE  'autovacuum: VACUUM%%'
-                     AND query NOT LIKE  'autovacuum: VACUUM ANALYZE%%'
-                     AND query NOT LIKE  '%%to prevent wraparound%%') AS vacuum,
-  count(*) FILTER (WHERE query LIKE  '%%to prevent wraparound%%') AS vacuum_freeze,
-  count(*) FILTER (WHERE query LIKE  'autovacuum: BRIN summarize%%') AS brin_summarize
-FROM pg_stat_activity
-WHERE query NOT LIKE '%%pg_stat_activity%%';
+    count(*) FILTER (
+WHERE
+    query LIKE  'autovacuum: ANALYZE%%') AS analyze,   count(*) FILTER (
+WHERE
+    query LIKE  'autovacuum: VACUUM ANALYZE%%') AS vacuum_analyze,   count(*) FILTER (
+WHERE
+    query LIKE  'autovacuum: VACUUM%%'                      
+    AND query NOT LIKE  'autovacuum: VACUUM ANALYZE%%'                      
+    AND query NOT LIKE  '%%to prevent wraparound%%') AS vacuum,   count(*) FILTER (
+WHERE
+    query LIKE  '%%to prevent wraparound%%') AS vacuum_freeze,   count(*) FILTER (
+WHERE
+    query LIKE  'autovacuum: BRIN summarize%%') AS brin_summarize 
+FROM
+    pg_stat_activity 
+WHERE
+    query NOT LIKE '%%pg_stat_activity%%';
 """,
     'DIFF_LSN': """
-SELECT pg_{0}_{1}_diff(CASE pg_is_in_recovery() WHEN true THEN pg_last_{0}_receive_{1}() ELSE pg_current_{0}_{1}() END, '0/0') as wal_writes ;
+SELECT
+    pg_{0}_{1}_diff(CASE pg_is_in_recovery() 
+        WHEN true THEN pg_last_{0}_receive_{1}() 
+        ELSE pg_current_{0}_{1}() 
+    END,
+    '0/0') as wal_writes ;
 """
 }
```

**Comment by Ferroin** [comment]
I guess I shouldn't be surprised that such a site exists.

Unless you have any objections, I'll add that as a commit.

**Comment by Ferroin** [comment]
This ideally needs tested before it gets merged.  As mentioned in the description, I don't have a working PGSQL installation immediately available that I can test against, so I'm probably not the best person to test it.

**Comment by paulfantom** [comment]
I don't see why it shouldn't work, but you are right, it should be tested first.

**Comment by Ferroin** [comment]
The patch itself wasn't exactly clean (lots of spurious whitespace, plus a spot where the formatting was off for the Python code), which, combined with me still not quite understanding exactly what all of this SQL code is doing, means I'm not quite as optimistic about it just working as you are.

**Comment by l2isbad** [comment]
Hi @anayrat 
Do you have time to help us on this?

**Comment by anayrat** [comment]
hi,

I'll look but formating made hard to read queries for human. Could you try with : 
http://sqlformat.darold.net
Thanks

**Comment by Ferroin** [comment]
@anayrat That actually does look better, thanks for the tip.

One issue with it though is that in the REPSLOT_FILES code, we end up with this:

    COALESCE(floor((pg_wal_lsn_diff (pg_current_wal_lsn (), slot.restart_lsn) - (pg_walfile_name_offset (restart_lsn)).file_offset) / (s.val)), 0) AS replslot_wal_keep

Which is still _way_ too long.  Any suggestions on how to split that line and keep it readable?

**Comment by Ferroin** [comment]
Reformatted as requested.  The above mentioned long line is still an issue however, so this should not be merged until that is dealt with.

**Comment by paulfantom** [comment]
@Ferroin or you can do this: http://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors

**Comment by Ferroin** [comment]
@paulfantom Except it's unfortunately about half way through a long multi-line string, so an inline comment isn't exactly easy to add, and working around that will arguably make things a lot harder to read.

I could instead update it to use implicit string concatenation, but I don't think flake8 will be happy about that either...

File: python.d/postgres.chart.py [hunk 1]
```diff
@@ -21,222 +21,358 @@
...
-  ORDER BY (pg_stat_file('pg_{0}/'||name)).modification, wal.name DESC) sub;
+    count(*) AS total_wal,
+    count(*) FILTER (
+    WHERE
+        TYPE = 'recycled') AS recycled_wal, count(*) FILTER (
+    WHERE
+        TYPE = 'written') AS written_wal
+FROM (
+    SELECT
+        wal.name, pg_ 0 file_name (
```
Comment (anayrat) [inline python.d/postgres.chart.py]: Last commit has broken queries. Here `pg_{0}file_name` has been changed to `pg_ 0 file_name`

**Comment by anayrat** [comment]
Look in my fork : https://github.com/anayrat/netdata/commits/postgres
I reverted last commit and made a new one which seems more readable.
I tested it on postgres from 9.3 to 10 without error.

**Comment by Ferroin** [comment]
@anyrat I see a couple of ways we can get this merged then:

1. I just close this PR and you open a new one with your branch.
2. I back out the last commit in my branch, merge the rest, and you open a PR with just the SQL fixes in it 
3. I back out the last commit in my branch and cherry-pick the commit from yours, and then merge it as part of this PR.

Given that the actual merge to the master branch results in a single squashed commit, I don't really have any preference as to how this gets resolved, so it's up to you.  The first and second approaches would require some effort on your end, while the third would require nothing from you other than double checking things here before they get merged.

**Comment by Ferroin** [comment]
@anayrat  That works too.  I'll have the PR updated shortly.

**Comment by Ferroin** [comment]
@l2isbad Could you take a look at this?

**Comment by Ferroin** [comment]
Well, that's ironic timing...

**Pull request merged**